src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -499,12 +500,63 @@ void StorageObjectStorage::truncate(
             getName(), path.path);
     }
 
-    StoredObjects objects;
-    for (const auto & key : configuration->getPaths())
+    if (configuration->partition_strategy && configuration->partition_strategy_type == PartitionStrategyFactory::StrategyType::HIVE) 
+    {
+        auto query_settings = configuration->getQuerySettings(context);
+        query_settings.throw_on_zero_files_match = false;
+        query_settings.ignore_non_existent_file = true;
+    
+        bool local_distributed_processing = distributed_processing;
+        if (context->getSettingsRef()[Setting::use_hive_partitioning])
```

Comment:
`use_hive_partitioning` is irrelevant when `partition_strategy='hive'`. Just hard code `false`.

Reply:
Fixed.